lib/pwwka/persistent_transmitter.rb

```diff
@@ -0,0 +1,165 @@
+require_relative "publish_options"
+
+begin  # optional dependency
+  require 'resque'
+  require 'resque-retry'
+rescue LoadError
+end
+
+module Pwwka
+  # Primary interface for sending messages.
+  #
+  # Example:
+  #
+  #     # Send a message, blowing up if there's any problem
+  #     Pwwka::Transmitter.send_message!({ user_id: @user.id }, "users.user.activated")
+  #
+  #     # Send a message, logging if there's any problem
+  #     Pwwka::Transmitter.send_message_safely({ user_id: @user.id }, "users.user.activated")
+  class PersistentTransmitter
+
+    extend Pwwka::Logging
+    include Pwwka::Logging
+
+    DEFAULT_DELAY_BY_MS = 5000
+
+    attr_reader :channel_connector
+
+    def initialize
+      @channel_connector = ChannelConnector.new(connection_name: "p: #{Pwwka.configuration.app_id} #{Pwwka.configuration.process_name}".strip)
+    end
+
+    def self.connection(routing_key, type: nil, headers: nil, message_id: :auto_generate)
+      connection = self.new
+      yeild(connection)
+    ensure
+      connection.channel_connector.connection_close
+    end
+
+
+    # Send an important message that must go through.  This method allows any raised exception 
+    # to pass through.
+    #
+    # payload:: Hash of what you'd like to include in your message
+    # routing_key:: String routing key for the message
+    # delayed:: Boolean send this message later
+    # delay_by:: Integer milliseconds to delay the message
+    # type:: A string describing the type.  This + your configured app_id should be unique to your entire ecosystem.
+    # message_id:: If specified (which generally you should not do), sets the id of the message.  If omitted, a GUID is used.
+    # headers:: A hash of arbitrary headers to include in the AMQP attributes
+    # on_error:: What is the behavior of
+    # - :ignore (aka as send_message_safely)
+    # - :raise
+    # - :resque -- use Resque to try to send the message later
+    # - :retry_async -- use the configured background job processor to retry sending the message later
+    #
+    # Returns true
+    #
+    # Raises any exception generated by the innerworkings of this library.
+    def self.send_message!(payload, routing_key,
+                           on_error: :raise,
+                           delayed: false,
+                           delay_by: nil,
+                           type: nil,
+                           message_id: :auto_generate,
+                           headers: nil)
+      if delayed
+        new.send_delayed_message!(*[payload, routing_key, delay_by].compact, type: type, headers: headers, message_id: message_id)
```

Comment:
You wouldn't want to call `new` here right?  That looks like it would create a new connection.  reusing the connection on `self` is probably what you want

Reply:
For this method, do you think we need it  and the `send_message_async`at all? its just a carry over from the original file and the way we intend to use this block depends on control of the instance anyway.

Reply:
true, go ahead and get rid of the things you don't need and we can build them back up in the future as we enhance the persistent transmitter to do more things.